### DISCUSSION TITLE
Backwards compatibility

### DISCUSSION THREAD

**Discussion by AndersRask** [pnp/powershell#83] [Q&A] [answered]
A question we get quite a lot from our developers and other customers is, what changes are required for existing scripts.
Some run in functions, some in runbooks, others in release pipelines.
- I guess PowerShell 7 is mandatory for this release?
- Are the modules implemented with breaking changes in mind, or is this designed as a new major version? What I am seeing so far testing, the later seems to be the case.
- What exactly does "archive old repro" mean in the roadmap? Will it still be available, just without bug fixes?

**Comment by erwinvanhunen** [comment, marked as answer]
Hi Anders,

1. No, PS 7 is not required. PS5 is also supported.
2. We treat it as a major release. And as a result, we -do- allow breaking changes. Check the changelog, where we list all the changes. We renamed some cmdlets, removed some cmdlets, but those are relatively small changes. The biggest and most major change is the way authentication works. We are now doing full OAuth based auth. See https://pnp.github.com/powershell. There are a few articles there that might help you.
3. We will move the old repro to a read-only state. E.g. it will stay available, but we will not release anymore bugfixes. Same goes for the legacy modules. They will stay available on the PowerShell Gallery, we will however not update them anymore.

**Comment by AndersRask** [comment]
Hi Erwin,

thx for the answer. 
Great that PS5 will also work, as this will ease the transition to the new platform. I guess there will be features in PS5 that will not work, regarding authentication?

With regards to old repro, I see that the help pages are already removed, at least when you type Get-Help <your legacy pnp cmdlet> -Online
Is this an error? It redirects to https://pnp.github.io/ 
Even though the module is depricated, the old documentation would be of great value when you work on old code -even when you figure out how to update it to new code. Also in other MS documentation you are able to select older platforms when looking at SDK's and similar, so this seems like a strange move to me :)
